dateutil/parser/_parser.py

```diff
@@ -1419,6 +1419,14 @@ def parse(self, tzstr):
                         signal = 1
                     used_idxs.append(i)
                     res.dstoffset = (res.stdoffset + int(l[i]) * signal)
+
+                # This was a made-up format that is not in normal use
+                warn(('Parsed time zone "%s"' % tzstr) +
+                     'is in a non-standard dateutil-specific format, which ' +
+                     'is now deprecated; support for parsing this format ' +
+                     'will be removed in future versions. It is recommended ' +
+                     'that you switch to a standard format like the GNU ' +
+                     'TZ variable format.', tz.DeprecatedTzFormatWarning)
```

Comment:
Suggestion: Include link to the GNU TZ variable format either withing dateutil documentation or to the format specification.

Reply:
That's a good idea, and one I considered. I think it is in the [tzstr documentation](https://dateutil.readthedocs.io/en/latest/tz.html#dateutil.tz.tzstr). I'm a bit hesitant to add it here, since I'm afraid of it going out of date. I'll merge this for now, we can tweak the message later.